CoffeeMachineSimulator/CoffeeMachineSimulator.Services/Services/CoffeeService.cs

```diff
@@ -0,0 +1,49 @@
+﻿using CoffeeMachineSimulator.Services.Interfaces;
+using CoffeeMachineSimulator.Services.Models;
+using FizzWare.NBuilder;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CoffeeMachineSimulator.Services.Services
+{
+    public class CoffeeService : ICoffeeService
+    {
+        private readonly List<CoffeeModel> coffeeModels;
+
+        public CoffeeService()
+        {
+            coffeeModels = Builder<CoffeeModel>.CreateListOfSize(10).Build().ToList();
+        }
+
+        public void AddCoffee(CoffeeModel coffeeToAdd)
+        {
+            if (coffeeToAdd == null) { throw new Exception("You should not add null entries!"); }
+            foreach(CoffeeModel coffee in coffeeModels)
+            { if (coffee.Id == coffeeToAdd.Id) { throw new Exception("Already exists an entry with the same Id"); } }
```

Comment:
Use coffeeModels.Any() method to check if there is any item that matches the criteria. https://stackoverflow.com/questions/879391/linq-any-vs-exists-whats-the-difference